app/views/handlers/primary_person_question.py

```diff
@@ -14,7 +14,16 @@ def _get_routing_path(self):
         return self.router.routing_path(section_id=self._current_location.section_id)
 
     def is_location_valid(self):
-        return self.router.can_access_location(self.parent_location, self._routing_path)
+        is_list_item_id_valid = (
+            self.current_location.list_item_id
+            in self._questionnaire_store.list_store[
+                self.current_location.list_name
+            ].items
+        )
```

Comment:
This was added because at the moment, you are able to manipulate the `list_item_id` when adding a primary person. Without this, you can change the list item id when adding a primary person and post the page which breaks the piping etc.

This check ensures that the `list_item_id` of the current location (add screen) exists in the list store (it should if it is valid because it is pregenerated for the primary person).

Reply:
Shouldn't it also check that the `list_item_id` is for the primary person?

Reply:
Good shout, will update.

Reply:
Updated